### PR TITLE
Tracking Security Fixes

### PR DESCRIPTION
**Pull request by Flickdm**
# Description

This change adds CVE Tracking to BaseTools, CryptoPkg, MdeModulePkg, NetworkPkg, UefiCpuPkg, this may be used to confirm if CVE's have been patched and contains references to relevant sources of information.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [X] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

N/A

## Integration Instructions

N/A

### CODE REVIEW

File: BaseTools/SecurityFixes.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+CVE_2017_5731__CVE_2017_5732__CVE_2017_5733__CVE_2017_5734__CVE_2017_5735:
+    commit_titles:
+        - # Upstream Patch from Tianocore Edk2
+            - "BaseTools: Add more checker in Decompress algorithm to access the valid buffer (CVE FIX)"
+    cve:
+        - CVE-2017-5731 # This is the only CVE that still is in use
```
Comment (Flickdm) [inline BaseTools/SecurityFixes.yaml:6]: discard the abandoned CVE's and change the name,
This could just be a note field

**Comment by makubacki** [comment]
Thanks for putting up this draft. Is documentation available describing more about how the files are generated and used?

**Comment by makubacki** [comment]
> Thanks for putting up this draft. Is documentation available describing more about how the files are generated and used?

I see https://github.com/microsoft/mu/pull/164 was posted, a link to that is what I was looking for.

**Comment by Flickdm** [comment]
Yeah sorry - I need to change my github notifications because I missed this. 
Yeah the link you mentioned is to the documentation for how these files are constructed / how to be used

**Comment by makubacki** [comment]
@Flickdm - What is the current plan for this?

**Pull request closed without merging**